docs/Guide.Jest.md

```diff
@@ -3,46 +3,88 @@ id: Guide.Jest
+
+![package.json](img/jest-guide/package_json.png)
+
+##### The [Jest configuration file](https://github.com/wix/Detox/blob/master/examples/demo-react-native-jest/e2e/config.json):
+
+- In the very least, verify the following fields: `testEnvironment`, `setupFilesAfterEnv`.
+
+  > Note: `setupFilesAfterEnv` was introduced in Jest 24.
+
+- **`reporters` array field, `verify` bolean field:**
+  We override Jest's default reporters so as to streamline logs of all flavours, in real-time - which isn't Jest's default, thus optimizing it for running e2e tests. This is optional: if you're OK with Jest's default logging, you can either remove `reporters` or set to the `"default"` item. See [Jest's docs](https://jestjs.io/docs/en/configuration#reporters-array-modulename-modulename-options) for more details.
+
+##### The [Jest custom initialization code](https://github.com/wix/Detox/blob/master/examples/demo-react-native-jest/e2e/init.js):
+
+- `beforeAll()`, `beforeEach()`, `afterAll()` should all call our custom Detox-Jest adapter's equivalent lifecycle methods.
+- The custom Detox-Jest adapter must be registered as a `jasmine` reporter (`jasmine.getEnv().addReporter()`).
+- Optional: An additional `trace-adapter` should be registered as a `jasmine` reporter, as well. This one takes care of logging on a per-spec basis (i.e. when an `it` starts and ends) — which Jest does not do by default. Should be used in conjunction with our custom reporters.
+
+![Streamlined output](img/jest-guide/streamlined_logging.png)
+
+### 4. Run some tests
+
+For example:
+
+```sh
+> react-native start
+...
+> detox test e2e --configuration ios.sim.debug
 ```
 
-### Writing Tests
+`detox init` generates default tests under the `e2e` folder. By now, you should be able to run them successfully, getting real-time, streamlined output and results.
+
+
+
+## Writing Tests
 
 There are some things you should notice:
 
 - Don't worry about mocks being used, Detox works on the compiled version of your app.
 - Detox exposes it's primitives (`expect`, `device`, ...) globally, it will override Jest's global `expect` object.
 
+
+
+## Parallelism
+
+Through Detox' cli, Jest can be started with [multiple workers](https://github.com/wix/Detox/blob/jest-boost/docs/APIRef.DetoxCLI.md#test) that run tests in simultaneously. In this mode, Jest effectively assigns one worker per each test file (invoking jasmine over it). In this mode, the per-spec logging offered by the trace-adapter mentioned earlier, does not necessarily make sense, as the workers' outputs get mixed up.
+
+Unfortunately, there's no way of telling whether parallelism has been enabled, at the adapter's scope (different JS contexts). If you find this way of logging unsuitable in a multiple-workers environment (typically an automated CI build), consider limiting the trace-adapter's registration in your test-init file (e.g. by using a dedicated environment variable).
```

Comment:
But we still can check (in the adapter code) for CLI flags like `max-workers`, at least to make an educated guess, right?

Reply:
Actually, I've tried to do a POC for that too, pls look:

https://github.com/wix/Detox/blob/1eaaf941bb3a534072c04373f7bd3f885496adbd/detox/runners/jest/impl/DetoxJestAdapter.js#L39-L49

Reply:
I was not aware of that the args pass through onto the various context. Will have look, thanks!